fedimint-core/src/runtime.rs

```diff
@@ -216,7 +218,7 @@ mod tests {
 
     #[tokio::test]
     async fn test_timeout_completed() {
-        let duration = Duration::from_secs(1);
+        let duration = Duration::from_millis(1);
```

Comment:
This will be flaky. You can not rely on one thing finishing before the other. With 1s duration, it would probably work reliably, but technically still be incorrect.

If you need to guarantee that one thing finished / or not before the other blocking on a receiver of a oneshot channel, and sending message to unblock it from the other side is the way to go.